src/tools/run-make-support/src/external_deps/c_build.rs

```diff
@@ -25,3 +28,33 @@ pub fn build_native_static_lib(lib_name: &str) -> PathBuf {
     llvm_ar().obj_to_ar().output_input(&lib_path, &obj_file).run();
     path(lib_path)
 }
+
+/// Builds a dynamic lib (`.dll` on Windows, `.dylib` on OSX and `.so` for Linux) with the given
+/// name.
```

Comment:
Suggestion: in the just-recently merged https://github.com/rust-lang/rust/pull/127960, I changed our dylib filename calculation to delegate to stdlib's provided info via `std::env::consts::{DLL_PREFIX, DLL_EXTENSION}` now that we can of course rely on std instead of having to roll our own in makefiles :3 Could you just say something to the effect of

> Builds a dynamic lib. The filename is computed on a target-dependent manner, relying on ``[`std::env::consts::DLL_PERFIX`]`` and ``[`std::env::consts::DLL_EXTENSION`]``.

That is, for example, on AIX the dylib extension is actually `.a` -- so we should just defer to stdlib constants and docs to be accurate :3